subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/LocalComponentDependencyMetadata.java

```diff
@@ -107,7 +107,7 @@ public String getDependencyConfiguration() {
 
     @Override
     public Set<ConfigurationMetadata> selectConfigurations(ComponentResolveMetadata fromComponent, ConfigurationMetadata fromConfiguration, ComponentResolveMetadata targetComponent, AttributesSchema attributesSchema) {
-        assert fromConfiguration.getHierarchy().contains(getOrDefaultConfiguration(moduleConfiguration));
+        // assert fromConfiguration.getHierarchy().contains(getOrDefaultConfiguration(moduleConfiguration));
```

Comment:
That's 100% correct. The rationale is that it has side effects in performance tests (shows hotspots where there are none in reality). I think such code belongs to debugging, this is why I only commented it, and not removed it. Also, there's a cost to assertions, which is bytecode length, which prevents the JIT from inlining methods.